normandy/recipes/validators.py

```diff
@@ -1,8 +1,33 @@
 import json
+import jsonschema
 
 from django.core.exceptions import ValidationError
 
 
+# Add path to required validator so we can get property name
+def _required(validator, required, instance, schema):
+    '''Validate 'required' properties.'''
+    if not validator.is_type(instance, 'object'):
+        return
+
+    for index, requirement in enumerate(required):
+        if requirement not in instance or instance[requirement] == '':
+            error = jsonschema.ValidationError(
+                'This field may not be blank.',
+                path=[requirement]
+            )
+            yield error
+
+
+# Construct validator as extension of Json Schema Draft 4.
+Validator = jsonschema.validators.extend(
```

Comment:
Maybe rename this to `JSONSchemaValidator` so it's a little clearer what it's validating and it the name won't conflict with any future validators we may write.